### ISSUE TITLE
Make fields immutable on edit

### ISSUE THREAD

**Issue by KaradzaJuraj** [cyclops-ui/cyclops#92]
Some fields shouldn't be editable, for example, the `name` field in the demo template.

We should add an option like `editable` to the `schema.json` file that tells us if the field will be editable through the module edit screen or present only when creating modules.

Cyclops needs to be able to parse this afterward.

**Comment by ak708** [comment]
@petar-cvit Can I work on this one?

**Comment by petar-cvit** [comment]
Hey @ak708, sure. Let me know if you have any questions

**Comment by ak708** [comment]
@petar-cvit Do you need the name field disabled in frontend alone, or does the values.schema.json need to be generated with an edit field?

**Comment by petar-cvit** [comment]
@ak708 can we add a field "immutable" set to true in the `values.schema.json`? If it is set, while editing the module the field should be shown, but should not be editable

**Comment by ak708** [comment]
@petar-cvit sure, but let me make sure i understand it right, we need functionality to edit templates from cyclops/templates repo, but name of template should be immutable and the json file must have immutable:"true" in the values.schema.json for that template right?

**Comment by petar-cvit** [comment]
@ak708 when you are creating a new module, you will get a form based on the template. That form is rendered based on the `values.schema.json` file. In the form you can input stuff for your configuration and you can edit them later on.

But, some fields might not be editable once they are set and we can call those fields immutable. When we render a form for module creation, you can change the field, but when you are editing the module the field cannot be changed. If you want to set a field to be immutable, you need to add `"immutable": true` to the field definition. For example if we want the replicas to be immutable, it would look like this:
```json
"replicas": {
  "description": "Number of replicas",
  "type": "integer",
  "immutable": true,
},
```

You will need to add the `immutable` field [here](https://github.com/cyclops-ui/cyclops/blob/0fafd86a3ee8f37934f017566c9f5ed53f794b74/cyclops-ctrl/internal/models/helm/helmschema.go#L8) and [map it](https://github.com/cyclops-ui/cyclops/blob/0fafd86a3ee8f37934f017566c9f5ed53f794b74/cyclops-ctrl/internal/mapper/helm.go#L72-L71) later on to propagate it to backend.

**Comment by ak708** [comment]
@petar-cvit alright

**Comment by ak708** [comment]
@petar-cvit , im pretty new to go, how do you run the cyclops-ctrl locally?, i tried the minikube through docker, but that launches its own instance of the gui, i also tried docker compose on cyclops-ctrl but that creates only a redis instance in docker which is also not on port 8888.

**Comment by petar-cvit** [comment]
@ak708 if you have a running cluster, you should be able to start the cyclops-ctrl by going into the `cyclops-ctrl` directory and running `go run -tags musl ./cmd/main`

**Comment by petar-cvit** [comment]
@ak708 do you have any updates on the issue? Let us know if you have any questions

**Comment by ak708** [comment]
@petar-cvit , im still working on it, was up to the neck till today, will get to it this week

**Comment by ak708** [comment]
@petar-cvit The fields in general have the immutable property and their inputs are disabled if its true, but do we need the same for the template edit at the top and the replicas and toggle switch?

**Comment by petar-cvit** [comment]
Hey @ak708, you dont have to implement it for a specific template, but just read the property 'immutable' and dosable the input.

Can you open a PR so we can discuss the solution? Thanks!

**Comment by petar-cvit** [comment]
implemented in #494
